scripts/translate_ascii.py

```diff
@@ -0,0 +1,59 @@
+import argparse
+import os
+import tempfile
+
+from scripts.aws.aws_helper import get_session, is_s3
+from scripts.files.fs import find_sidecars, write_all
+from scripts.files.fs_s3 import bucket_name_from_path, list_uri
+from scripts.gdal.gdal_helper import run_gdal
+
+
+def main() -> None:
+    """Translates ascii files in a given path to tiffs and writes to target,
+
+    Arguments:
+    --source - local or s3 path to source data
+    --target - local or s3 path to write converted tiffs
+
+    examples:
+        python translate_ascii.py --source s3://linz-elevation-staging/test/sample-tests/ --target /tmp/
+        python translate_ascii.py --source ./tests/data/ --target /tmp/
+    """
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--source", dest="source", required=True, help="S3 path to the input ascii files")
+    parser.add_argument("--target", dest="target", required=True, help="Output location path")
+    arguments = parser.parse_args()
+
+    source = arguments.source
+
+    with tempfile.TemporaryDirectory() as tmp_path:
+        if is_s3(source):
+            # list ascii files from s3
+            s3 = get_session(source).client("s3")
+            asc_files = list_uri(source, s3, extension=".asc")
+            # reformat to full s3 path
+            asc_files = [f"s3://{bucket_name_from_path(source)}/{file}" for file in asc_files]
+            # download ascii files
+            asc_files = write_all(inputs=asc_files, target=tmp_path, concurrency=25)
+
+        else:
+            # list ascii files from local
+            asc_files = [os.path.join(source, file) for file in os.listdir(os.path.abspath(source)) if file.endswith(".asc")]
+
+        tiffs = []
+        for asc in asc_files:
+            # translate from ascii to geotiff using GDAL
+            tiff = os.path.join(tmp_path, f"{asc.split('/')[-1].split('.')[0]}.tiff")
+            command = ["gdal_translate", "-of", "GTiff", asc, tiff]
```

Comment:
What does this output as? do we need a `-co COMPRESS=lzw` `-co NUM_THREADS=all_cpus` etc?

Reply:
I've added both `-co COMPRESS=lzw` and `-co NUM_THREADS=all_cpus` this script runs before standarding so I was intending just to convert the ascii file to tiff and let standardising handle the rest?

Reply:
Ideally we should never store anything uncompressed, converting from ASC to uncompressed tiff is a improvement, but still should atleast be `lzw` 

I thinkg there are a few standard parameters we need to always add to `gdal_translate` like the `NUM_THREADS` and defaulting to lzw for compression seems easy.